specs/ProgrammaticSaveAs.md

```diff
@@ -0,0 +1,422 @@
+                        CHECK_FAILURE(args->put_Kind(dialog.selectedKind));
+                        CHECK_FAILURE(args->put_AllowReplace(dialog.allowReplace));
+                    }
+                    else
+                    {
+                        // Save As cancelled from this customized dialog
+                        CHECK_FAILURE(args->put_Cancel(TRUE));
+                    }
+                };
+
+                wil::com_ptr<ICoreWebView2Deferral> deferral;
+                CHECK_FAILURE(args->GetDeferral(&deferral));
+
+                m_appWindow->RunAsync(
+                    [deferral, showCustomizedDialog]()
+                    {
+                        showCustomizedDialog();
+                        CHECK_FAILURE(deferral->Complete());
+                    });
+                return S_OK;
+            })
+            .Get(),
+        &m_saveAsRequestedToken);
+    
+    MessageBox(
+        m_appWindow->GetMainWindow(), L"Event Handler Added", L"Info",MB_OK);
+    return true;
+}
+```
+### Programmatic Save As
+Call RequestSaveAs method to trigger the programmatic save as.
+```c++
+
+bool ScenarioSaveAs::ProgrammaticSaveAs()
+{
+    if (!m_webView2_20)
+        return false;
+    m_webView2_20->RequestSaveAs(
+        Callback<ICoreWebView2RequestSaveAsCompletedHandler>(
+            [this](HRESULT errorCode, COREWEBVIEW2_SAVE_AS_REQUESTED_RESULTS result) -> HRESULT
+            {
+                // Show RequestSaveAs returned result, optional
+                MessageBox(
+                    m_appWindow->GetMainWindow(),
+                    (L"Save As " + saveAsResultString[result]).c_str(), L"Info", MB_OK);
+                return S_OK;
+            })
+            .Get());
+    return true;
+}
+```
+
+## .Net/ WinRT
+### Add the Event Handler
+This example hides the default save as dialog and shows a customized dialog.
+```c#
+
+void AddEventHandlerExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    webView.CoreWebView2.SaveAsRequested -= WebView_SaveAsRequested;
+    MessageBox.Show("Event Handler Added", "Info");
+}
+
+void WebView_SaveAsRequested(object sender, CoreWebView2SaveAsRequestedEventArgs args)
+    {
+    // Hide the system default save as dialog.
+    args.SuppressDefaultDialog = true;
+
+    // Developer can obtain a deferral for the event so that the CoreWebView2
+    // doesn't examine the properties we set on the event args until
+    // after the deferral completes asynchronously.
+    CoreWebView2Deferral deferral = args.GetDeferral();
+
+    // We avoid potential reentrancy from running a message loop in the event
+    // handler. Show the customized dialog later when complete the deferral
+    // asynchronously.
+    System.Threading.SynchronizationContext.Current.Post((_) =>
+    {
+        using (deferral)
+        {
+            // This is a customized dialog example, the constructor returns after the 
+            // dialog interaction is completed by the end user.
```

Comment:
(C++ comment doesn't apply to C# sample)

```suggestion
            // This is a customized dialog example.
```

Reply:
Please fix.